### PR TITLE
deps: fix dependency build

### LINKED ISSUE #2296: Get new kokoro builds passing

### PR DESCRIPTION
**Pull request by kolea2**
Fixes #2296

### CODE REVIEW

File: bigtable-client-core-parent/bigtable-client-core/pom.xml [hunk 1]
```diff
@@ -100,10 +105,23 @@ limitations under the License.
...
+            <artifactId>google-cloud-core</artifactId>
+        </dependency>
         <dependency>
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.threeten</groupId>
+            <artifactId>threetenbp</artifactId>
+            <version>1.4.0</version>
```
Comment (igorbernstein2) [inline bigtable-client-core-parent/bigtable-client-core/pom.xml]: Please promote this to a variable

File: bigtable-client-core-parent/bigtable-hbase/pom.xml [hunk 2]
```diff
@@ -64,16 +64,66 @@ limitations under the License.
...
+            <artifactId>google-auth-library-credentials</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.grpc</groupId>
+            <artifactId>grpc-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.google.api</groupId>
+            <artifactId>gax</artifactId>
+            <version>1.48.1</version>
```
Comment (igorbernstein2) [inline bigtable-client-core-parent/bigtable-hbase/pom.xml]: please use variable

File: bigtable-dataflow-parent/bigtable-beam-import/pom.xml [hunk 3]
```diff
@@ -80,23 +73,48 @@ limitations under the License.
...
+      <artifactId>beam-sdks-java-core</artifactId>
+      <version>${beam.version}</version>
     </dependency>
     <dependency>
-      <groupId>org.hamcrest</groupId>
-      <artifactId>hamcrest-library</artifactId>
-      <version>${hamcrest.version}</version>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
+      <version>1.7.29</version>
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-beam-import/pom.xml]: extract to beam-slf4j.version

File: bigtable-dataflow-parent/bigtable-beam-import/pom.xml [hunk 4]
```diff
@@ -80,23 +73,48 @@ limitations under the License.
...
+      <artifactId>avro</artifactId>
+      <version>1.8.2</version>
+    </dependency>
+    <dependency>
+      <groupId>com.google.cloud.bigtable</groupId>
+      <artifactId>bigtable-hbase-1.x-shaded</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.beam</groupId>
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-beam-import/pom.xml]: lets group this together with the main beam dep above

File: bigtable-dataflow-parent/bigtable-hbase-beam/pom.xml [hunk 5]
```diff
@@ -205,6 +214,13 @@ limitations under the License.
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <configuration>
+                    <usedDependencies>org.apache.commons:commons-compress,org.apache.beam:beam-sdks-java-io-hbase</usedDependencies>
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-hbase-beam/pom.xml]: Please add a comment why these are here:
- compress is for security fixes
- beam-sdks-java-io-hbase is for the HBaseCoderProviderRegistrar

Also please separate the beam-sdks-java-io-hbase dependency above from the security comment and add a similar comment about the HBaseCoderProviderRegistrar

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-hadoop/pom.xml [hunk 6]
```diff
@@ -71,6 +78,18 @@ limitations under the License.
       <artifactId>metrics-core</artifactId>
       <version>${dropwizard.metrics.version}</version>
     </dependency>
+    <dependency>
+      <groupId>org.apache.hbase</groupId>
+      <artifactId>hbase-common</artifactId>
+      <version>${hbase.version}</version>
+    </dependency>
```
Comment (igorbernstein2) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-hadoop/pom.xml]: Please keep hbase & hadoop deps together

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml [hunk 7]
```diff
@@ -283,11 +283,42 @@ limitations under the License.
...
+
         <dependency>
             <groupId>org.apache.hbase</groupId>
             <artifactId>hbase-client</artifactId>
             <version>${hbase.version}</version>
         </dependency>
+        <dependency>
+            <groupId>com.google.code.findbugs</groupId>
+            <artifactId>jsr305</artifactId>
+            <version>${jsr305.version}</version>
```
Comment (igorbernstein2) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml]: all of these deps should be scoped to test

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml [hunk 8]
```diff
@@ -283,11 +283,42 @@ limitations under the License.
...
+        <dependency>
+            <groupId>org.apache.hbase</groupId>
+            <artifactId>hbase-common</artifactId>
+            <version>${hbase.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.hamcrest</groupId>
+            <artifactId>hamcrest</artifactId>
+            <version>${hamcrest.version}</version>
+        </dependency>
```
Comment (igorbernstein2) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml]: I think we should stay consistent with the rest of the modules and check testing frameworks at the end

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml [hunk 9]
```diff
@@ -283,11 +283,42 @@ limitations under the License.
...
         <dependency>
```
Comment (igorbernstein2) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml]: can we move module dependencies to the top?

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml [hunk 10]
```diff
@@ -359,6 +384,13 @@ limitations under the License.
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <configuration>
+                    <usedDependencies>org.apache.hbase:hbase-testing-util,org.apache.hbase:hbase-server</usedDependencies>
```
Comment (igorbernstein2) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml]: please add a comment that this necessary to force hbase version in the common tests?

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x/pom.xml [hunk 11]
```diff
@@ -32,17 +32,73 @@ limitations under the License.
...
+    </dependency>
+    <dependency>
+      <groupId>com.google.cloud.bigtable</groupId>
+      <artifactId>bigtable-client-core</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>com.google.protobuf</groupId>
+      <artifactId>protobuf-java</artifactId>
+      <version>3.9.1</version>
```
Comment (igorbernstein2) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x/pom.xml]: variable please

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x-integration-tests/pom.xml [hunk 12]
```diff
@@ -317,13 +317,49 @@ limitations under the License.
...
+            <groupId>org.apache.hbase</groupId>
+            <artifactId>hbase-client</artifactId>
+            <version>${hbase.version}</version>
+            <scope>test</scope>
+        </dependency>
 
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>12.0.1</version>
```
Comment (igorbernstein2) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x-integration-tests/pom.xml]: pulll this out as hbase-guava.version

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x-integration-tests/pom.xml [hunk 13]
```diff
@@ -317,13 +317,49 @@ limitations under the License.
...
         <dependency>
             <groupId>junit</groupId>
             <artifactId>junit</artifactId>
             <version>${junit.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.hamcrest</groupId>
+            <artifactId>hamcrest</artifactId>
+            <version>2.2</version>
```
Comment (igorbernstein2) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x-integration-tests/pom.xml]: please use the already defined variable

Comment (igorbernstein2) [reply bigtable-hbase-2.x-parent/bigtable-hbase-2.x-integration-tests/pom.xml]: also do you need hamcrest-core as well? I think you can remove it as a dep and from the unused list

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/pom.xml [hunk 14]
```diff
@@ -36,25 +36,83 @@ limitations under the License.
...
+    </dependency>
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+    </dependency>
+
+    <dependency>
+      <groupId>io.dropwizard.metrics</groupId>
+      <artifactId>metrics-core</artifactId>
+      <version>3.1.2</version>
```
Comment (igorbernstein2) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x/pom.xml]: extract version

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/pom.xml [hunk 15]
```diff
@@ -36,25 +36,83 @@ limitations under the License.
...
+    </dependency>
+
+    <!-- Test dependencies-->
+    <dependency>
+      <groupId>org.hamcrest</groupId>
+      <artifactId>hamcrest-core</artifactId>
+      <version>1.3</version>
```
Comment (igorbernstein2) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x/pom.xml]: use variable & maybe use hamcrest instead to be consistent with other modules

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x-integration-tests/pom.xml (hunk 13 above)
Comment (kolea2) [reply bigtable-hbase-2.x-parent/bigtable-hbase-2.x-integration-tests/pom.xml]: it's required, build fails without it

**Pull request merged**